packages/publisher/github/src/PublisherGithub.ts

```diff
@@ -110,21 +116,36 @@ export default class PublisherGithub extends PublisherBase<PublisherGitHubConfig
                 return done();
               }
             }
-            await github.getGitHub().repos.uploadReleaseAsset({
-              owner: config.repository.owner,
-              repo: config.repository.name,
-              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-              release_id: release!.id,
-              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-              url: release!.upload_url,
-              // https://github.com/octokit/rest.js/issues/1645
-              data: (await fs.readFile(artifactPath)) as unknown as string,
-              headers: {
-                'content-type': mime.lookup(artifactPath) || 'application/octet-stream',
-                'content-length': (await fs.stat(artifactPath)).size,
-              },
-              name: artifactName,
-            });
+            try {
+              const { data: uploadedAsset } = await github.getGitHub().repos.uploadReleaseAsset({
+                owner: config.repository.owner,
+                repo: config.repository.name,
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                release_id: release!.id,
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                url: release!.upload_url,
+                // https://github.com/octokit/rest.js/issues/1645
+                data: (await fs.readFile(artifactPath)) as unknown as string,
+                headers: {
+                  'content-type': mime.lookup(artifactPath) || 'application/octet-stream',
+                  'content-length': (await fs.stat(artifactPath)).size,
+                },
+                name: artifactName,
+              });
+              if (uploadedAsset.name !== sanitizedArtifactName) {
+                // There's definitely a bug with GitHub.sanitizeName
+                console.warn(logSymbols.warning, chalk.yellow(`Expected artifact's name to be '${sanitizedArtifactName}' - got '${uploadedAsset.name}'`));
+              }
+            } catch (err) {
+              // If an asset with that name already exists, it's either a bug with GitHub.sanitizeName
+              // where it did not sanitize the artifact name in the same way as GitHub did, or there
+              // was simply a race condition with uploading artifacts with the same name
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              if (err instanceof RequestError && err.status === 422 && (err.response?.data as any)?.errors?.[0].code === 'already_exists') {
+                console.error(`Asset with name '${artifactName} already exists - there may be a bug with GitHub.sanitizeName`);
```

Comment:
We should make it clear in logging that this is a Forge internal util rather than something on GitHub's side.